### PR TITLE
Add test for GRE tunnel with xml

### PR DESCRIPTION
**Pull request by asmorodskyi**

### CODE REVIEW

**Comment by asmorodskyi** [comment]
@jlausuch  it is not finished yet, also I need to check but this is what I have up to now ...

File: tests/wicked/advanced_ref.pm [hunk 1]
```diff
@@ -34,29 +33,35 @@ sub create_tunnel_with_commands {
...
     my ($self) = @_;
 
-    record_info('Test 1', 'Create a gre interface from legacy ifcfg files');
+    record_info('Test 1', 'Create a GRE interface from legacy ifcfg files');
     $self->create_tunnel_with_commands("gre1", "gre", "24");
     mutex_wait('test_1_ready');
-    assert_script_run("ifdown gre1");
-    # Placeholder for Test 2: Create a GRE interface from Wicked XML files
+    assert_script_run("ip addr flush dev gre1");
+    assert_script_run("ip tunnel delete gre1");
```
Comment (jlausuch) [inline tests/wicked/advanced_ref.pm]: you could also group this commands in a cleanup function

File: tests/wicked/advanced_sut.pm [hunk 2]
```diff
@@ -47,79 +60,83 @@ sub setup_tunnel {
...
 sub setup_bridge {
-    my ($self, $config, $dummy, $br_name, $dummy_name) = @_;
+    my ($self, $config, $dummy) = @_;
     my $local_ip = $self->get_ip(no_mask => 1, is_wicked_ref => 0, type => 'host');
     assert_script_run("sed \'s/ip_address/$local_ip/\' -i $config");
     assert_script_run("cat $config");
     assert_script_run("cat $dummy");
-    assert_script_run("ifup $br_name");
-    assert_script_run("ifup $dummy_name");
+    assert_script_run("ifup br0");
```
Comment (jlausuch) [inline tests/wicked/advanced_sut.pm]: ok, since we only have 1 bridge type and it's always br0

File: tests/wicked/advanced_sut.pm [hunk 3]
```diff
@@ -47,79 +60,83 @@ sub setup_tunnel {
...
 sub run {
     my ($self) = @_;
     my %results;
     my $iface = script_output('ls /sys/class/net/ | grep -v lo | head -1');
 
     $self->before_scenario('Test 1', 'Create a gre interface from legacy ifcfg files');
-    my $config = '/etc/sysconfig/network/ifcfg-gre1';
-    $self->get_from_data('wicked/ifcfg-gre1_', $config, add_suffix => 1);
+    my $config = '/etc/sysconfig/network/ifcfg-gre1';    
+    $self->get_from_data('wicked/ifcfg-gre1_sut', $config);
```
Comment (jlausuch) [inline tests/wicked/advanced_sut.pm]: I would even go further and delete the suffix in the files, but maybe for another PR.

File: tests/wicked/advanced_ref.pm (hunk 1 above)
Comment (asmorodskyi) [reply tests/wicked/advanced_ref.pm]: will do this after I will get a green bubble with what I have now :)

File: tests/wicked/advanced_sut.pm (hunk 3 above)
Comment (asmorodskyi) [reply tests/wicked/advanced_sut.pm]: +1 for another PR, too many things happening in PR which has a humble name "Add test for GRE tunnel with xml" :)

**Comment by asmorodskyi** [comment]
proof runs : 
http://kimball.arch.suse.de/tests/1421
http://kimball.arch.suse.de/tests/1420

**Comment by asmorodskyi** [comment]
@jlausuch with cleanup I think it is a valid idea but we can do it later. let's merge what we have so we can both continue

**Pull request merged**